### ISSUE TITLE
typescript needs to be updated to 1.49.3

### ISSUE THREAD

**Issue by ericwill** [eclipse-che/che#16186]
A new version of the typescript plugin is available upstream at:
https://github.com/Microsoft/vscode

**Comment by tsmaeder** [comment]
With Typescript 1.49.3, the typescript service crashes immediately upon startup. I thought this was a problem with creating the log directory (the /home/theia/.theia directory is not properly created on the sidecar). However, it still works with the same error message when running in a self-hosting workspace. Checking memory requirements next.

**Comment by tsmaeder** [comment]
So after properly creating the log directory, turns out the tsserver crashes without writing any logs. Further hacking of the typescript extensions reveals that it cannot load the module `--max-oldspace-size=3096`. Very odd!

**Comment by tsmaeder** [comment]
I was able to confirm that this seems to be a problem with how `nexe`, the tool we use to compile the plugin host executable handles arguments to `child_process.fork(...)`: when I run the following code snippet in node 12.18.2:
```
cp.fork('sub.js', ['subarg'], {
    silent: false,
    execArgv: [ '--max-old-space-size=3072', '--use-bundled-ca']
})
```

the script `sub.js` is executed. When I compile the script with `nexe main.js` the resulting executable fails with the same error output as what I'm observing when running the typescript extension:

```
internal/modules/cjs/loader.js:969
  throw err;
  ^

Error: Cannot find module '--max-old-space-size=3072'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
    at internal/main/run_main_module.js:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

**Comment by tsmaeder** [comment]
I turns out when I combine the right nodejs with the right nexe version, everything works. It would be nice to have a move obvious way to find out which version of nodejs is "supported".

**Comment by benoitf** [comment]
closed by accident ?

**Comment by tsmaeder** [comment]
Yes, wanted to close another issue...too many tabs exception.

**Comment by ericwill** [comment]
This was fixed in 7.24.
